appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -1944,6 +1956,40 @@ private Option elementToOption(Element field, String value) {
     );
   }
 
+  /**
+   * Returns the associated helper key if the element has an @Asset annotation. Null otherwise.
+   *
+   * @param elem the Element which represents a function (for return types) or a parameter.
+   * @param type the TypeMirror representing the type of that element.
+   * @return the associated helper key if the element has an @Asset annotation.
+   */
+  private HelperKey<Integer> hasAssetsHelper(Element elem, TypeMirror type) {
```

Comment:
I'm curious how this approach will work with extensions. If you and I both create an extension with a filter, can't we end up with the same index? When we load these two extensions into App Inventor, do we end up selecting the same (potentially incorrect) filter?

I recall we decided to drop filtering for now, but you may want to leave a TODO here to think about that.